src-java/blue-green/src/main/java/org/openkilda/bluegreen/ZkWatchDog.java

```diff
@@ -61,38 +61,39 @@ public ZkWatchDog(String id, String serviceName, String connectionString,
             signal = Signal.NONE;
         }
         this.signal = signal;
-        initWatch();
     }
 
     @Override
-    void validateNodes() throws KeeperException, InterruptedException {
+    void validateNodes() throws KeeperException, InterruptedException, IOException {
         super.validateNodes();
         ensureZNode(serviceName, id, SIGNAL);
         ensureZNode(DEFAULT_BUILD_VERSION.getBytes(), serviceName, id, BUILD_VERSION);
+        nodesValidated = true;
```

Comment:
Terrible logic tangling between this class and it's parent. I can't propose anything to fix it right now... except removing this `nodesValidated` flag.

Reply:
I want to solve following problem. 
We need to check that we connected to ZK and it works fine. We need to check not one, but on every action (when we receive tupple, when we reading massage from kafka, etc)
So this check must be fast. Current check method checks
1. that we connected to zk
2. that we able to create nodes and default nodes were created

If we will check just 1 it would be useless. Because clients wouldn't able to work with Zookeeper. 
Maybe I can rename `isConnectionAlive` method to something else. like `isZooKeeperZeady ToWork`